src/box/iproto.cc

```diff
@@ -1283,6 +1290,18 @@ iproto_connection_input_buffer(struct iproto_connection *con)
 	return new_ibuf;
 }
 
+static inline bool
+iproto_flushed(struct iproto_connection *con)
+{
+	struct obuf *obuf = con->wpos.obuf;
+	struct obuf_svp *begin = &con->wpos.svp;
+	struct obuf_svp *end = &con->wend.svp;
+
+	if (con->wend.obuf != obuf)
+		return (begin->used == obuf->used) && (end->used == 0);
+	return (begin->used == end->used);
```

Comment:
AFAIU if both buffers are empty then `wpos` and `wend` must point to the same buffer. That is, the following check should be enough, no?

```C
return con->wpos.obuf == con.wend.obuf &&
       con->wpos.svp.used == con->wend.svp.used;
```

Reply:
Yes, it's true. Thanks!